core/variant/variant_internal.h

```diff
@@ -1537,7 +1537,7 @@ struct VariantTypeConstructor {
 	}
 
 	_FORCE_INLINE_ static void type_from_variant(void *p_value, void *p_variant) {
-		*((T *)p_value) = VariantInternalAccessor<T>::get(reinterpret_cast<Variant *>(p_variant));
+		*((T *)p_value) = *reinterpret_cast<Variant *>(p_variant);
```

Comment:
Is there a reason for the C-style cast instead of reinterpret_cast everywhere ?
```c++
*reinterpret_cast<T *>p_value = *reinterpret_cast<Variant *>(p_variant);
```

Reply:
I don't know the answer to the question :-) but because that cast isn't added by this PR, I don't think this should block this PR.

There's loads and loads of casts in GDExtension, and more than one bug has been caused by using the inappropriate cast for the situation, so perhaps we need an issue/PR that's dedicated to sorting out all the different casts?